shared/actions/profile/index.js

```diff
@@ -224,7 +224,9 @@ function outputInstructionsActionLink(): Constants.OutputInstructionsActionLink
 }
 
 function* _onAppLink(action: AppLink): SagaGenerator<any, any> {
-  const match = action.payload.link.match(/^https:\/\/keybase\.io\/(\w+)$/)
+  // TODO: When https://github.com/facebook/react-native/issues/16434
+  // is fixed, use window.URL.
+  const match = action.payload.link.match(/^https:\/\/keybase\.io\/(\w+)(\?[^?]*)(#[^#]*)?$/)
```

Comment:
Should add `/i` flag to this so it's case insensitive, and also have (http(s)) optional.

Perhaps `/^(http(s)?:\/\/)?keybase\.io\/(\w+)(\?[^?]*)?(#[^#]*)?$/i`

Reply:
Also in testing I noticed that navigating to `https://keybase.io/chris` works, but `https://keybase.io/Chris` leads to an empty profile. Not sure what's causing that...

Reply:
Decided to instead pull in a URL parsing library, per @chrisnojima. Also added some tests. Should handle pretty much anything we throw at it now.